# Problem Pre-Defined Set (If isRandom is False)

# robot_number: 4 
# robot_position/x: [0, 1, 2, 3]
# robot_position/y: [0, 3, 2, 0] 
# robot_velocity: [3, 3, 5, 5] # UGV Speed = 2.0 m/s; UAV Speed = 8.0 m/s
# robot_battery_level: [80, 20, 30, 30] # b = robot battery level in mAh
# robot_battery_discharge_rate: [0.2, 0.3, 0.6, 0.6] # w = robot battery discharge rate
# robot_lambda: [2, 2, 1, 1]  # Robot

# task_number: 10
# task_position/x: [0, 1, 2, 1, 4, 5, 1, 3, 8, 2]
# task_position/y: [0, 1, -2, 3, -4, 5, 6, 7, 4, 9]
# task_velocity/x: [1, 1, 1, 1, 1, 1, 1, 1, 1, 1]
# task_velocity/y: [1, 1, 1, 1, 1, 1, 1, 1, 1, 1]
# task_priority: [1, 1, 1, 1, 1, 1, 1, 1, 1, 1]
# task_operation_time: [10, 10, 10, 5, 5, 8, 8, 4, 1, 2]

robot_number: 3 
robot_position/x: [5, 8, 8]
robot_position/y: [1, 2, 4] 
robot_velocity: [7.7841, 13.7504, 7.2414] # UGV Speed = 2.0 m/s; UAV Speed = 8.0 m/s
robot_battery_level: [90.6713, 63.9078, 3.0128] # b = robot battery level in mAh
robot_battery_discharge_rate: [0.7967, 0.5837, 0.8075] # w = robot battery discharge rate
robot_lambda: [1.5939, 2.1652, 4.6304]  # Robot

task_number: 4
task_position/x: [4, 6, 6, 1]
task_position/y: [7, 11, 10, 15]
task_velocity/x: [-2.29357603931582, -3.37534327743011, 1.07729236492944, -0.521987609839201]
task_velocity/y: [2.93366560763503, 0.221657230651301, 3.38156709571015, -1.52854502300792]
task_priority: [2.52340371029146, 5.05027634784029, 2.05264856608328, 9.75438365172804]
task_operation_time: [4.79907592547609, 8.47750680271680, 7.52684431446140, 2.31600439984813]

time_horizon: 4

# Parameter for Randomization (If isRandom is True)
max_robot_number: 10
max_task_number: 20

max_field_dimension_x: 15
max_field_dimension_y: 15


